src/ppx/command_line_parser.cpp

```diff
@@ -118,25 +133,73 @@ std::optional<CommandLineParser::ParsingError> CommandLineParser::Parse(int argc
         }
         name = name.substr(2);
 
+        std::string_view value = "";
         if (i + 1 < args.size()) {
-            std::string_view nextElem = ppx::string_util::TrimBothEnds(args[i + 1]);
+            auto nextElem = ppx::string_util::TrimBothEnds(args[i + 1]);
             if (!StartsWithDoubleDash(nextElem)) {
-                // We found an option with a parameter.
-                mOpts.AddOption(name, nextElem);
+                // The next element is a parameter for the current option
+                value = nextElem;
                 ++i;
-                continue;
             }
         }
-        // There is no parameter so it's likely a flag
-        if (name.substr(0, 3) == "no-") {
-            mOpts.AddOption(name.substr(3), "0");
+
+        if (auto error = AddOption(name, value)) {
+            return error;
+        }
+    }
+
+    std::vector<std::string> configJsonPaths;
+    configJsonPaths = mOpts.GetOptionValueOrDefault("config-json-path", configJsonPaths);
+    for (const auto& jsonPath : configJsonPaths) {
+        std::ifstream f(jsonPath);
+        if (f.fail()) {
+            return "Invalid --config-json-path: " + jsonPath;
+        }
+        nlohmann::json data = nlohmann::json::parse(f);
+        if (auto error = AddJsonOptions(data)) {
+            return error;
+        }
+    }
+
+    return std::nullopt;
+}
+
+std::optional<CommandLineParser::ParsingError> CommandLineParser::AddJsonOptions(const nlohmann::json& jsonConfig)
```

Comment:
Overall: don't assume anything regarding the json config layout.
The top-level object could be an array, flag values could be arrays when not expected, or even objects, etc.
Please use the `is_array()`, `is_object()`, etc to validate the structure when loading it.
Reading `it->value()` is fine as we delegate the parsing to the command_line utility, but the rest should be validated.